dali/operators/generic/resize/tensor_resize.h

```diff
@@ -0,0 +1,131 @@
+#include "dali/operators/image/resize/resize_base.h"
+#include "dali/operators/image/resize/tensor_resize_attr.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace tensor_resize {
+
+
+template <typename Backend>
+class TensorResize : public Operator<Backend>
+                   , protected ResizeBase<Backend> {
+ public:
+  explicit TensorResize(const OpSpec &spec);
+
+ protected:
+  int NumSpatialDims() const { return resize_attr_.spatial_ndim_; }
+  int FirstSpatialDim() const { return resize_attr_.first_spatial_dim_; }
+
+  bool CanInferOutputs() const override { return true; }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override;
+
+  void RunImpl(Workspace &ws) override;
+
+  void PrepareParams(const ArgumentWorkspace &ws, const TensorListShape<> &input_shape) {
+    int nsamples = input_shape.num_samples();
+    resize_attr_.PrepareResizeParams(spec_, ws, input_shape);
+
+    // TODO(janton) : implement 1D
+    if (NumSpatialDims() == 1) {
+      throw std::invalid_argument("1D resizing is not yet implemented.");
+    }
+
+    if (NumSpatialDims() < 1) {
+      throw std::invalid_argument(
+          "TensorResize expects at least one spatial dimension to be resized");
+    }
+
+    if (NumSpatialDims() > 3) {
+      throw std::invalid_argument(make_string(
+          "TensorResize can not resize more than 3 consecutive spatial dimensions. "
+          "The input may contain extra leading and trailing dimensions which are not resized.\n\n"
+          "Example 1: Input shape (2, 2, 4, 3, 2), Output shape (2, 2, 5, 5, 2). Only the 3rd and "
+          "4th dimensions are resized (2 spatial dimensions)\n\n"
+          "Example 2: Input shape (2, 2, 4, 3, 2), Output shape (2, 3, 4, 5, 2). Only the 2nd and "
+          "4th dimensions are resized, but we can only consider a block of consecutive dimensions "
+          "so we include the 3rd dimension as well (3 spatial dimensions).\n\nGot: ",
+          resize_attr_.spatial_ndim_, " spatial dimensions starting at dim ",
+          resize_attr_.first_spatial_dim_));
+    }
+
+    assert(NumSpatialDims() >= 1 && NumSpatialDims() <= 3);
+    assert(FirstSpatialDim() >= 0);
+    resample_params_.resize(nsamples * NumSpatialDims());
+    resampling_attr_.PrepareFilterParams(spec_, ws, nsamples);
+    resampling_attr_.GetResamplingParams(make_span(resample_params_),
+                                         make_cspan(resize_attr_.params_));
+  }
+
+  void InitializeBackend();
+
+  USE_OPERATOR_MEMBERS();
+  std::vector<kernels::ResamplingParams> resample_params_;
+  using Operator<Backend>::RunImpl;
+
+  TensorResizeAttr resize_attr_;
+  ResamplingFilterAttr resampling_attr_;
+};
+
+template <typename Backend>
+TensorResize<Backend>::TensorResize(const OpSpec &spec)
+    : Operator<Backend>(spec), ResizeBase<Backend>(spec), resize_attr_(spec) {
+  resample_params_.resize(num_threads_);
```

Comment:
Why do we need that resize here? You do next the:
```
resample_params_.resize(nsamples * NumSpatialDims());
```

Reply:
Copied from the other resize, but I don't think it makes any sense here.